rfd/0204-aws-iam-roles-anywhere.md

```diff
@@ -0,0 +1,769 @@
+│ └────────────────────────────────────────┘ │ │                     │  
+└────────────────────────────────────────────┘ └─────────────────────┘  
+```
+
+##### How IAM Roles Anywhere work with Teleport
+
+Teleport periodically syncs IAM Roles Anywhere Profiles as an Application resource into Teleport, which allows you to define RBAC policies on them.
+
+##### RBAC
+Re-using the Teleport Application resources gets us the following features out of the box:
+- support for just-in-time Access Requests
+- support for Access Lists
+
+AWS Applications are created from a Profile and the Profile Tags are mapped into Application Labels.
+To grant access to a Profile, users must be allowed to access the Application using `app_labels`.
+
+When accessing AWS using a Profile/Application, users must also have access to the IAM Role they want to assume.
+AWS validates that only Profile's allowed Roles are accessible and Teleport validates that only `aws_role_arn` (either in traits or explicit in Teleport Roles) can assume that IAM Role.
+
+In order to assume a given Role, it must be present on both:
+- Profile's allowed IAM Roles
+- Teleport's `aws_role_arn` list for the User.
+
+As an example, assuming the following IAM Roles Anywhere Profile:
+```yaml
+name: ProfileA
+tags:
+  - Team: ABC
+  - Env: Prod
+roles:
+  - MyRoleA
+  - MyRoleB
+```
+
+A Teleport Application will be created with the following metadata:
+```yaml
+kind: app_server
+metadata:
+  labels:
+    Team: ABC
+    Env: Prod
+  name: ProfileA
+spec:
+  app:
+    kind: app
+    metadata:
+      labels:
+          Team: ABC
+          Env: Prod
+      name: ProfileA
+    spec:
+      aws:
+        roles_anywhere:
+          allowed_roles_arn:
+          - arn:aws:iam::123456789012:role/MyRoleA
+          - arn:aws:iam::123456789012:role/MyRoleB
+          profile_arn: arn:aws:rolesanywhere:eu-west-2:123456789012:profile/ac1f655b-aaaa-aaaa-aaaa-aaaaaaaaaaaa
+      cloud: AWS
+      integration: raa
+      uri: https://console.aws.amazon.com
+    version: v3
+version: v3
+```
+
+And the following Teleport Role would grant access:
+```yaml
+kind: role
+metadata:
+  description: AWS Access for ProfileA
+  name: access-profile-a
+spec:
+  allow:
+    app_labels:
+      Team: ABC
+      Env: Prod
+    aws_role_arns:
+    - arn:aws:iam::123456789012:role/MyRoleA
+    - arn:aws:iam::123456789012:role/MyRoleB
+version: v7
+```
+
+##### New Resources
+The following IAM resources must be be created in your AWS account.
+
+**IAM Roles Anywhere Trust Anchor** that trusts Teleport's AWS Roles Anywhere Certificate Authority.
+This ensures that IAM trusts Teleport as a Roles Anywhere certificate issues.\
+This certificate can be found at `curl https://<proxy-url>/v1/webapi/auth/export?type=awsra` or `tctl auth export --type awsra`.
+
+A new **IAM Role** which allows Teleport to query existing IAM Roles Anywhere Profiles, in order to sync them into Teleport.
```

Comment:
so basically this is a bootstrap role that uses the same anchor and profile right?

Reply:
Yes, this allows the Profile->AppServer sync process to run.

In the future, we could use the IAM Roles Anywhere integration just for auto discovery of SSH/DB/Kube; and completely ignore the Profile->AppServer sync.
For that scenario, you would still set up the Trust Anchor but create a different bootstrap role for the particular scenario you are interested in (either one IAM Role for each type of resource or for the entire DiscoveryService, we'll think about that when we get there).